metrics/metrics-dropwizard/src/main/java/org/apache/james/metrics/dropwizard/DropWizardTimeMetric.java

```diff
@@ -20,17 +20,54 @@
 package org.apache.james.metrics.dropwizard;
 
 import org.apache.james.metrics.api.TimeMetric;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import com.codahale.metrics.Timer;
+import com.google.common.base.Preconditions;
 
 public class DropWizardTimeMetric implements TimeMetric {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(DropWizardTimeMetric.class);
+
+    static class DropWizardExecutionResult implements ExecutionResult {
+        private final String name;
+        private final long elaspedInNanoSeconds;
+        private final long p99InNanoSeconds;
+
+        DropWizardExecutionResult(String name, long elaspedInNanoSeconds, long p99InNanoSeconds) {
+            Preconditions.checkArgument(elaspedInNanoSeconds > 0);
+            Preconditions.checkArgument(p99InNanoSeconds > 0);
+
+            this.name = name;
+            this.elaspedInNanoSeconds = elaspedInNanoSeconds;
+            this.p99InNanoSeconds = p99InNanoSeconds;
+        }
+
+        @Override
+        public long elaspedInNanoSeconds() {
+            return elaspedInNanoSeconds;
+        }
+
+        @Override
+        public ExecutionResult logWhenExceedP99(long thresholdInNanoSeconds) {
+            Preconditions.checkArgument(thresholdInNanoSeconds > 0);
+            if (elaspedInNanoSeconds > p99InNanoSeconds && elaspedInNanoSeconds > thresholdInNanoSeconds) {
```

Comment:
s/&&/|| ?

Reply:
No I want to log p99 if it is high enough. I don't care about 5ms p99... It's a && (or the threshold makes no sens IMO)